pinot-common/src/main/java/org/apache/pinot/pql/parsers/pql2/ast/FilterKind.java

```diff
@@ -19,33 +19,41 @@
 package org.apache.pinot.pql.parsers.pql2.ast;
 
 public enum FilterKind {
-  AND,
-  OR,
-  NOT,
-  EQUALS,
-  NOT_EQUALS,
-  GREATER_THAN,
-  GREATER_THAN_OR_EQUAL,
-  LESS_THAN,
-  LESS_THAN_OR_EQUAL,
-  LIKE,
-  BETWEEN,
-  RANGE,
-  IN,
-  NOT_IN,
-  REGEXP_LIKE,
-  IS_NULL,
-  IS_NOT_NULL,
-  TEXT_MATCH,
-  JSON_MATCH;
+  AND(false, false),
```

Comment:
Is this change related? We should not rely on type `FilterKind` for the purpose of this PR

Reply:
Yes, we use FilterKind in PredicateComparisonRewriter. It's used to determine we need to convert to EQUALS expression. For example:
`WHERE col1 and col2` should be converted
 `WHERE col1 BETWEEN 10 AND 20` should not be converted.

Reply:
I checked the usage of `isPredicate()` and it is actually not needed. See the comments in `PredicateComparisonRewriter`.